API/src/bank-accounts/hendlers/index.ts

```diff
@@ -0,0 +1,209 @@
+import { getConnection } from '../../data_base/data_base';
+
+
+
+//--------------------------------------GET ALL ACTIONS----------------------------------
+
+async function getActions(req, res, next) {
+    const accountNumber = req.params.accountNumber
+
+    const result = await getActionsHandler(accountNumber)
+    if (!result) throw new Error("Data not found!")
+    console.log(accountNumber)
+    return res.status(200).json(result);
+}
+
+async function getActionsHandler(accountNumber) {
+    const query = getActionsQuery();
+    const [result] = await getConnection().execute(query, [accountNumber]);
+    console.log(result)
+    console.log("******************ACTIONS FATCHED FROM DB**********************")
+    return result;
+}
+
+const getActionsQuery = () => {
+    //     return `SELECT 
+    //     actions.id as action_id,
+    //     account_number,
+    //     deposit_sum,
+    //     deposit_date,
+    //     withdrawal_sum,
+    //     withdrawal_date,
+    //     hypothec_sum,
+    //     commission,
+    //     payments,
+    //     hypothec_date
+    // FROM
+    //     bank_accounts.actions
+    //         LEFT JOIN
+    //     bank_accounts.accounts ON bank_accounts.accounts.id = bank_accounts.actions.account_id
+    //         LEFT JOIN
+    //     bank_accounts.deposit ON bank_accounts.deposit.id = bank_accounts.actions.deposit_id
+    //         LEFT JOIN
+    //     bank_accounts.hypothec ON bank_accounts.hypothec.id = bank_accounts.actions.hypothec_id
+    //         LEFT JOIN
+    //     bank_accounts.withdrawal ON bank_accounts.withdrawal.id = bank_accounts.actions.withdrawal_id
+    // WHERE (account_number = ?)`;
+
+    return `SELECT 
+actions.id as action_id,
+account_number,
+deposit_sum,
+deposit_date,
+withdrawal_sum,
+withdrawal_date,
+hypothec_sum,
+commission,
+payments,
+hypothec_date
+FROM
+bank_accounts.actions
+    LEFT JOIN

```

Comment:
i think that planning such structure can be painfull in the future
no need to maintain too much tables
you could add all the actions with thier additional info inside 1 table

Reply:
JOIN cost performance, and for simple query look how much JOINS you are making